plugin/shell_plugin.go

```diff
@@ -35,10 +37,16 @@ func (s Sheller) Run(context Context, phase Phase) error {
 	//get all host ip
 	masterIP := context.Cluster.Spec.Masters.IPList
 	nodeIP := context.Cluster.Spec.Nodes.IPList
-	hostIP := append(masterIP, nodeIP...)
+	allHostIP := append(masterIP, nodeIP...)
+	//get on
+	on := context.Plugin.Spec.On
+	if on != "" {
```

Comment:
What about on : role=master or on : role=node

Reply:
have two pr，next pr will support it